server/src/test/java/org/elasticsearch/common/time/Iso8601ParserTests.java

```diff
@@ -0,0 +1,417 @@
+            .optionalStart()
+            .appendZoneOrOffsetId()
+            .optionalEnd()
+            .optionalStart()
+            .appendOffset("+HHmm", "Z")
+            .optionalEnd()
+            .toFormatter(Locale.ROOT)
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        // just checking timezones/ids here
+        assertEquivalent("2023-01-01T12:00:00", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00Z", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00UT", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00UTC", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00GMT", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00+00", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00-00", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00+05", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00+0500", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00+05:00", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00+05:00:30", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00-07", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00-0715", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00-07:15", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00UTC+05:00", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00GMT-09:45:30", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00Zulu", mandatoryFormatter);
+        assertEquivalent("2023-01-01T12:00:00Europe/Paris", mandatoryFormatter);
+
+        assertEquivalentFailure("2023-01-01T12:00:00+5", mandatoryFormatter);
+        assertEquivalentFailure("2023-01-01T12:00:00-7", mandatoryFormatter);
+        assertEquivalentFailure("2023-01-01T12:00:00InvalidTimeZone", mandatoryFormatter);
+
+        DateTimeFormatter allFieldsOptional = new DateTimeFormatterBuilder().appendValue(YEAR, 4, 4, SignStyle.EXCEEDS_PAD)
+            .optionalStart()
+            .appendLiteral('-')
+            .appendValue(MONTH_OF_YEAR, 2)
+            .optionalStart()
+            .appendLiteral('-')
+            .appendValue(DAY_OF_MONTH, 2)
+            .optionalStart()
+            .appendLiteral('T')
+            .appendValue(HOUR_OF_DAY, 2)
+            .optionalStart()
+            .appendLiteral(':')
+            .appendValue(MINUTE_OF_HOUR, 2)
+            .optionalStart()
+            .appendLiteral(':')
+            .appendValue(SECOND_OF_MINUTE, 2)
+            .optionalEnd()
+            .optionalEnd()
+            .optionalEnd()
+            .optionalEnd()
+            .optionalEnd()
+            .optionalStart()
+            .appendZoneOrOffsetId()
+            .optionalEnd()
+            .optionalStart()
+            .appendOffset("+HHmm", "Z")
```

Comment:
The old parser technically allows multiple timezones to be specified: `2023-01-01T12:00:00Europe/Paris+0400`, and it'll use the first one. Do we still need to accept these odd strings?

Reply:
Is that valid ISO8601? If not, then it doesn't seem like we need to support it.

Reply:
It's not, and the behaviour here is ambiguous anyway - so we can probably drop this

Reply:
If it's not ISO8601, I'm +1 for dropping